packages/lesswrong/components/editor/RateLimitWarning.tsx

```diff
@@ -9,25 +9,30 @@ const RateLimitWarning = ({lastRateLimitExpiry, rateLimitMessage}: {
   rateLimitMessage?: string
 }) => {
 
-  // "fromNow" makes for a more human readable "how long till I can comment/post?".
-  // moment.relativeTimeThreshold ensures that it doesn't appreviate unhelpfully to "now"
-  moment.relativeTimeThreshold('ss', 0);
-  // format momentJS fromNow to say "3 seconds" or "3 minutes" rather than 3s or 3m
-  // moment.updateLocale('en', {
-  //   relativeTime: {
-  //     s: 'a few seconds', ss: '%d seconds',
-  //     m: 'a minute',      mm: '%d minutes',
-  //     h: 'an hour',       hh: '%d hours',
-  //     d: 'a day',         dd: '%d days',
-  //     w: 'a week',        ww: '%d weeks'
-  //   }
-  // });
-  const fromNow = moment(lastRateLimitExpiry).fromNow(true)
+  const getTimeUntilNextPost = () => {
+    const diffInSeconds = moment(lastRateLimitExpiry).diff(moment(), 'seconds')
+    const diffInMin = moment(lastRateLimitExpiry).diff(moment(), 'minutes')
+    const diffInHours = moment(lastRateLimitExpiry).diff(moment(), 'hours')
+    const diffInDays = moment(lastRateLimitExpiry).diff(moment(), 'days')
+    const diffInWeeks = moment(lastRateLimitExpiry).diff(moment(), 'weeks')
+    if (diffInSeconds < 60) {
+      return `${diffInSeconds} second${diffInSeconds > 1 ? 's' : ''}`
+    }
+    if (diffInMin < 60) {
+      return `${diffInMin} minute${diffInMin > 1 ? 's' : ''}}`
+    }
+    if (diffInHours < 24) {
+      return `${diffInHours} hour${diffInHours > 1 ? 's' : ''}`
+    }
+    if (diffInDays < 7) {
+      return `${diffInDays} day${diffInDays > 1 ? 's' : ''}`
+    }
+    return `${diffInWeeks} week${diffInWeeks > 1 ? 's' : ''}`
+  }
 
-  let message = `Please wait ${fromNow} before posting again. ${rateLimitMessage ?? ''}`
+  let message = `Please wait ${getTimeUntilNextPost()} before posting again. ${rateLimitMessage ?? ''}`
   if (isEAForum) {
-    const diffInMin = moment(lastRateLimitExpiry).diff(moment(), 'minutes')
-    message = `You've written more than 3 comments in the last 30 min. Please wait ${diffInMin} min before commenting again. ${rateLimitMessage ?? ''}`
+    message = `You've written more than 3 comments in the last 30 min. Please wait ${getTimeUntilNextPost()} min before commenting again. ${rateLimitMessage ?? ''}`
```

Comment:
Can you replace `min` with `minutes` to match the new function's copy, and remove the extra `min` after `${getTimeUntilNextPost()}`?